### PR TITLE
Additional OID DB improvments

### PR DESCRIPTION
**Pull request by npmccallum**

### CODE REVIEW

File: const-oid/oiddbgen/src/root.rs [hunk 1]
```diff
@@ -1,80 +1,38 @@
...
-                    .entry(spec)
-                    .or_insert_with(Spec::default)
-                    .push(Node::new(obji, name));
-            }
-        }
+    pub fn add(
+        &mut self,
+        spec: impl Deref<Target = str>,
+        name: impl Deref<Target = str>,
+        obid: impl Deref<Target = str>,
```
Comment (tarcieri) [inline const-oid/oiddbgen/src/root.rs:16]: Implicit deref coercion should make the `impl Deref`s unnecessary:

```suggestion
        spec: &str,
        name: &str,
        obid: &str,
```

Comment (npmccallum) [reply const-oid/oiddbgen/src/root.rs:16]: It forces `main.rs` to take a reference. But I don't care either way.

Comment (npmccallum) [reply const-oid/oiddbgen/src/root.rs]: Fixed.

Comment (tarcieri) [reply const-oid/oiddbgen/src/root.rs]: IMO it's pretty much the main reason the `Deref` trait exists: so "smart pointer" types that impl `Deref` can be treated as regular references

File: const-oid/oiddbgen/src/asn1.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
+use std::collections::{BTreeMap, HashSet};
+
+use regex::Regex;
+
+#[derive(Clone, Debug)]
+pub struct Asn1Parser(String, BTreeMap<String, (Option<String>, Option<String>)>);
```
[the remark this answers is not in the document]
Comment (npmccallum) [reply const-oid/oiddbgen/src/asn1.rs]: Fixed.

**Review by tarcieri — APPROVED** [review]

**Pull request merged**